plumbing/format/idxfile/writer.go

```diff
@@ -0,0 +1,132 @@
+package idxfile
+
+import (
+	"bytes"
+	"math"
+	"sort"
+
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/utils/binary"
+)
+
+type object struct {
```

Comment:
we already have `Entry` in this package with this exact same struct, maybe you can use that?